devimint/src/vars.rs

```diff
@@ -0,0 +1,131 @@
+#![allow(non_snake_case)]
+
+use std::path::{Path, PathBuf};
+
+pub trait ToEnvVar {
+    fn to_env_value(&self) -> String;
+}
+
+macro_rules! declare_vars {
+    ($name:ident = ($($args:tt)*) =>
+        {
+            $($env_name:ident : $env_ty:ty = $env_value:expr;)*
+        }
+    ) => {
+        pub struct $name {
+            $(
+                #[allow(unused)]
+                pub $env_name: $env_ty
+            ),*
+        }
+
+        impl $name {
+            pub async fn new_vars($($args)*) -> ::anyhow::Result<Self> {
+                $(let $env_name: $env_ty = $env_value.into();)*
+                Ok(Self {
+                    $($env_name),*
+                })
+            }
+
+            pub fn vars<'a>(&'a self) -> impl Iterator<Item = (&'static str, String)> {
+                vec![$((stringify!($env_name), $crate::vars::ToEnvVar::to_env_value(&self.$env_name))),*].into_iter()
+            }
+        }
+    };
+}
+
+impl ToEnvVar for PathBuf {
+    fn to_env_value(&self) -> String {
+        self.as_os_str().to_str().expect("must be utf8").to_owned()
+    }
+}
```

Comment:
we can avoid this trait (use ToString instead) if we implement a wrapper `PathBuf` say `Utf8PathBuf`. we can also remove `utf8` function then.